dat/events/scavenger_event.lua

```diff
@@ -0,0 +1,2077 @@
+local function pilot_askFire(edata, npc_id)
+    local approachtext = _([[Would you like to do something with this scavenger?
+
+Pilot credentials:]])
+    local upgradechoice = "Settle debts"
+    local upgradequery = "Are you sure you want to settle {name}'s debt of {credits}?"
+    local upgrade_amount = 250e3 -- negligible effect, a red herring and credit sink, but the escort changes ships/outfits
+    local limit_ships = spob.cur():shipsSold()
+
+    -- if there is a shipyard, upgrade for more to buy a new ship from this shipyard
+    -- (the player is hoping for a dream ship, let's have the odds favorable)
+    if next(limit_ships) ~= nil then
+        upgrade_amount = 1e6 -- should be enough for a dreamship after a couple of tries with few experience points
+    end
+
+    if edata.debt < 1e6 then
+        upgradechoice = "Sponsor ugrades"
+        upgradequery = "Are you sure you want to sponsor {name} for upgrades, costing you up to {upgrade_amount}?"
+    end
+    local n, _s =
+        tk.choice("", getOfferText(approachtext, edata), upgradechoice, _("Abandon scavenger"), _("Do nothing"))
+    if
+        n == 1 and
+            tk.yesno(
+                "",
+                fmt.f(
+                    upgradequery,
+                    {name = edata.name, credits = fmt.credits(edata.debt), upgrade_amount = fmt.credits(upgrade_amount)}
+                )
+            )
+     then
+        if edata.debt >= 1e6 then
+            if player.pilot():credits() < edata.debt then
+                -- not enough credits
+                tk.msg(
+                    _("Not Enough Money"),
+                    _([["I can't fly on dreams and steam. Try again when you have enough money."]])
+                )
+                return
+            end
+            player.pay(-edata.debt)
+            edata.total_cost = edata.total_cost + edata.debt
+            edata.debt = 0
+        else
+            -- upgrade the escort
+            if player.pilot():credits() < upgrade_amount then
+                -- not enough credits, upgrade for less
+                upgrade_amount = math.floor(player.pilot():credits() / 2)
+            end
+            -- pay
+            player.pay(-upgrade_amount)
+            edata.total_cost = edata.total_cost + upgrade_amount
+            -- sell current ship
+            local _n, ship_escrow = edata.ship:price()
+
+            -- do the balancing
+            edata.wallet = edata.wallet + upgrade_amount + ship_escrow - edata.debt
+            edata.debt = 0
+            -- get a new ship
+            _createReplacementShip(edata, limit_ships)
+            -- apply upgrade experience bonus
+            edata.wallet = edata.wallet + math.floor(upgrade_amount * edata.experience * 0.05)
+            edata.last_sentiment = "content"
+        end
+    elseif
+        n == 2 and tk.yesno("", fmt.f(_("Are you sure you want to get rid of {name}? This cannot be undone."), edata))
+     then
+        local k = #mem.persons
```

Comment:
This `k` is defined, but not actually used and shadowed by the `k` in the for loop.